client/lib/app/util/parse-file-type.js

```diff
@@ -10,13 +10,47 @@ var TYPES = {
   cmmn: 'http://www.omg.org/spec/CMMN'
 };
 
+var defintionsNoPrefix = /[<\s]+(?:Definitions|definitions)\s+/,
```

Comment:
So what you're trying to do is to extract the root namespace from the first tag?

Test `<definitions ` => `xmlns="{uri}"`
Test `<foo:definitions` => `xmlns:foo="{uri}"`

?

Reply:
Exactly.

Reply:
The logic you've implemented looks quite complicated and is prone to [catastropic backtracking](https://www.regular-expressions.info/catastrophic.html). Why don't you use an actual XML parser such as [saxen](https://github.com/nikku/saxen) to start parsing the input until the desired root tag is found (from which you may extract the namespace data). This way we allow an XML parser to do the heavy lifting, handle special chars, broken input, etc...

Using saxen it could looks something like this (untested _!_):

```javascript
var {
  Parser
} = require('saxen');

function getRootNamespace(xml) {

  let namespace;

  const parser = new Parser();

  parser.on('openTag', function(elementName, getAttrs) {

    // only parse first tag
    parser.stop();

    // bpmn:definitions
    // dmn:Definitions
    if (elementName.indexOf(':')) {
      let [ prefix, name ] = elementName.split(/:/);

      if (name.toLowerCase() === 'definitions') {
        namespace = getAttrs()[`xmlns:${prefix}`];
      }
    }

    // definitions
    // Defintitions
    if (elementName.toLowerCase() === 'definitions') {
      namespace = getAttrs().xmlns;
    }
  });
  
  parser.parse(xml);

  return namespace;
}


var namespace = getRootNamespace(`<bpmn:Definitions ...>`);
```

As an example `'` is a valid attribute separator in XML, too and we'd need to account for it if we don't rely on an XML parser.